### ISSUE TITLE
libtrorrent (qbittorrent v5.0.1) create empy folder files when torrent paused

### ISSUE THREAD

**Issue by axet** [arvidn/libtorrent#7799]
**Please provide the following information**

libtorrent version (or branch): 2.0.10 (debian)

platform/architecture: gnu/linux debian trixie

compiler and compiler version: debian

qbittorrent create empty folders and files for stopped torrents on app restart. Only happened when torrents has empty files (0 size).

1) download torrent
2) stop it
3) exit qbittorrent
4) remove downloaded files from hdd
4) start qbittorrent

You will see bunch of folders created, torrent not active stop with error: "missing files". To reproduce you need torrent with 0 size files in it. All folders with 0 size files, will be recreated.

**Comment by arvidn** [comment]
yes, this is by design. libtorrent has traditionally treated adding a *new* torrent to a session the same as starting up an old session and adding back the old torrents. For this reason, adding a torrent for the first time behaves similar to adding back an old torrent.

In other words, it's not recorded anywhere that these folders and empty files have already been created, so they will be created every time.

Perhaps there should be a way for the client to indicate to libtorrent whether empty files should be created or not. The client (qbt in this case) will know whether it's a new torrent being added, or whether it was part of the old session.

**Comment by arvidn** [comment]
if the file priority of the empty files are 0, they shouldn't be re-created btw
